packages/authentication/docs/controller-functions.md

```diff
@@ -0,0 +1,63 @@
+## Endpoint definitions
+
+The following decorated controller functions demos the endpoints described at the beginning of markdown file [authentication-system](../authentication-system.md).
+
+Please note how they are decorated with `@authenticate()`, the syntax is:
+`@authenticate(<strategy_name>, {action: <action_name>, session: <enabled_or_not>})`
+
+- /login
+
+```ts
+class UserController {
+  @post('/login', ...APISpec)
+  login() {
+    // static route
+  }
+}
+```
+- /loginWithFB
+
+```ts
+class UserController {
+  @post('/loginWithFB', ...APISpec)
+  @authenticate('oath2.fb', {action: 'login', session: false})
+  loginWithFB() {}
```

Comment:
If we use `login with facebook`, the browser will be redirected to facebook login page to perform the oAuth 2.0 `implicit` or `authorization code` flow. Both of them require a callback URL. When the controller method is decorated as shown, what logic should be in `loginWithFB` method? Is it the callback function or just a place holder route to excise the FB login flow? I imagine that `/loginWithFB` will initiate the FB login.

Reply:
> Is it the callback function or just a place holder route to excise the FB login flow?

The second one "a place holder route to excise the FB login flow".

We find it pretty hard to illustrate the oauth+session strategies without implementing an example, so I created a story to explorer it then revisit the design here and update accordingly.

https://github.com/strongloop/loopback-next/issues/2624

Reply:
My proposal would be defining a separate callback endpoint, e.g. `login/facebook` and `login/facebook/callback`